python/rmm/tests/test_rmm.py

```diff
@@ -942,3 +943,46 @@ def test_rmm_device_buffer_copy(cuda_ary, make_copy):
     result = db_copy.copy_to_host()
 
     np.testing.assert_equal(expected, result)
+
+
+@pytest.mark.parametrize("level", rmm.logging_level)
+def test_valid_logging_level(level):
+    with warnings.catch_warnings():
```

Comment:
suggestion (blocking): Rather than using `warnings.catch_warnings`, prefer `pytest.warns`. The former simply suppresses warnings, while the latter will actually require the warning be thrown. For that to work here you will have to condition the check on level being high enough, but this can be done pretty easily because the `logging_level` should be an integer enum and therefore you can do comparisons like `if level > MIN_COMPILED_LEVEL` where `MIN_COMPILED_LEVEL = logging_level.DEBUG` or so. 

You may find https://github.com/rapidsai/cudf/blob/branch-23.10/python/cudf/cudf/testing/_utils.py#L433 helpful.

Reply:
This requires me to somehow get the compiled level into Python. I prefer not to plumb that.

Reply:
Ah good point. Yes that's true, if we don't already have that available then we can punt on this change.